services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java

```diff
@@ -3551,24 +3554,37 @@ private void safeDeleteRTTopic(String clusterName, String storeName, Store store
             clusterName,
             controllerClientEntry.getKey(),
             storeResponse.getError());
-        return;
+        return false;
       }
-      canDeleteRT = !Version.containsHybridVersion(storeResponse.getStore().getVersions());
-    }
-    if (canDeleteRT) {
-      String rtTopicToDelete = Version.composeRealTimeTopic(storeName);
-      truncateKafkaTopic(rtTopicToDelete);
-      for (ControllerClient controllerClient: controllerClientMap.values()) {
-        controllerClient.deleteKafkaTopic(rtTopicToDelete);
-      }
-      // Check if there is incremental push topic exist. If yes, delete it and send out to let other controller to
-      // delete it.
-      String incrementalPushRTTopicToDelete = Version.composeSeparateRealTimeTopic(storeName);
-      if (getTopicManager().containsTopic(pubSubTopicRepository.getTopic(incrementalPushRTTopicToDelete))) {
-        truncateKafkaTopic(incrementalPushRTTopicToDelete);
-        for (ControllerClient controllerClient: controllerClientMap.values()) {
-          controllerClient.deleteKafkaTopic(incrementalPushRTTopicToDelete);
-        }
+      if (Version.containsHybridVersion(storeResponse.getStore().getVersions())) {
+        LOGGER.warn(
+            "Topic {} cannot be deleted yet because the store {} has at least one hybrid version",
+            rtTopicName,
+            storeName);
+        return false;
+      }
+    }
+    return true;
+  }
+
+  private void deleteRealTtimeTopic(String clusterName, String storeName) {
+    Map<String, ControllerClient> controllerClientMap = getControllerClientMap(clusterName);
+    String rtTopicToDelete = Version.composeRealTimeTopic(storeName);
+    deleteRTTopicFromAllFabrics(rtTopicToDelete, controllerClientMap);
+    // Check if there is incremental push topic exist. If yes, delete it and send out to let other controller to
+    // delete it.
+    String incrementalPushRTTopicToDelete = Version.composeSeparateRealTimeTopic(storeName);
+    if (getTopicManager().containsTopic(pubSubTopicRepository.getTopic(incrementalPushRTTopicToDelete))) {
+      deleteRTTopicFromAllFabrics(incrementalPushRTTopicToDelete, controllerClientMap);
+    }
+  }
+
+  private void deleteRTTopicFromAllFabrics(String topic, Map<String, ControllerClient> controllerClientMap) {
+    truncateKafkaTopic(topic);
```

Comment:
Did you verify if `controllerClientMap` excludes the local fabric? i.e. we might be making unnecessary duplicate calls to `truncateKafkaTopic`.

Reply:
I don't think local fabric is excluded from `controllerClientMap`. removing the extra call

Reply:
However, it was being done like that on left side too. I basically do not intend to change the code logic of this part, I am only trying to refactor this part to extract out `isRTTopicDeletionPermittedByAllControllers`

Reply:
I put it back because a test `testHybridStoreToBatchOnly` was failing in `Assert.assertTrue(veniceAdmin.isTopicTruncated(rtTopic))`. It was not truncating the topic because `multiClusterConfigs.getControllerConfig(clusterName).getChildDataCenterControllerUrlMap()` was empty